actions/parse-push/src/main.ts

```diff
@@ -1,27 +1,149 @@
 import * as core from '@actions/core'
+import * as exec from '@actions/exec'
 import * as gh from '@actions/github'
 import * as Webhooks from '@octokit/webhooks'
 import * as fs from 'fs'
 import * as path from 'path'
 
-// /**
-//  * Extract the list of exercises from a list of files.
-//  *
-//  * @param files contains a list of files
-//  */
-// async function getExercises(files: string[]): Promise<Set<string>> {
-//     const re = /exercises\/([a-z-]*)\//
-//     let exs = new Set<string>()
-
-//     for (const f of files) {
-//         const m = re.exec(f)
-//         if (m) {
-//             exs.add(m[0])
-//         }
-//     }
-
-//     return exs
-// }
+// TODO: Create a common package with types
+type DispatchPayload = Record<string, Commit>
+
+type Commit = {
+    message: string
+    slug?: string
+    pull_request_html_url?: string
+    new_cases?: string[]
+}
+
+/**
+ * Extract the list of exercises from a list of files.
+ *
+ * @param files contains a list of files
+ */
+async function getSlugs(files: string[]): Promise<Set<string>> {
+    const re = /exercises\/([a-z-]*)\//
+    let exs = new Set<string>()
+
+    for (const f of files) {
+        const m = re.exec(f)
+        if (m) {
+            exs.add(m[0])
+        }
+    }
+
+    return exs
+}
+
+/**
+ * Parse the files modified in the commit and extract the exercise slug.
+ *
+ * Assumption: Commits may only modify one exercise.
+ *             This usually holds true but there may be exceptions.
+ *             In those cases, show a warning and return an empty slug.
+ *
+ * @param commit
+ */
+async function getSlug(commit: any): Promise<string> {
+    const candidates = await getSlugs(commit.modified)
+
+    // Assumption: Commits may only modify one exercise
```

Comment:
I've already broken that. Is there a reason why we want to enforce this?

Reply:
No. It's just what a brief look at the PR view of prob-specs revealed to be the case for most PRs. Should be easy enough to fix, just have to iterate through files individually.